src/components/EditUserModal/EditUserModal.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState } from "react";
+import { Modal, Input, Form } from "antd";
+
+import { updateUserById } from "../../api/Users.api";
+
+import { IFullUser } from "../../models/types";
+
+interface ModalProps {
+  isOpen: boolean;
+  closeModal: () => void;
+  user: IFullUser;
+}
+
+const EditUserModal: React.FC<ModalProps> = ({ isOpen, closeModal, user }) => {
+  const [confirmLoading, setConfirmLoading] = useState(false);
+  const [formInstance] = Form.useForm();
+
+  const closeAndReset = () => {
+    closeModal();
+    formInstance.resetFields();
+  };
+
+  const submitModal = async () => {
+    setConfirmLoading(true);
+
+    try {
+      const values = await formInstance.validateFields();
+      await updateUserById(user.id, values);
+      setConfirmLoading(false);
```

Comment:
Please move the logic to a hook that uses react-query mutation,
you can see an example here
https://github.com/marik22312/radiosavta-backoffice/blob/master/src/components/EditProgramTimes/EditProgramTimes.tsx#L49

and onSucces, please invalidate the query by usersId

Reply:
This will also provide you with `isLoading` stuff